### PR TITLE
Parsing of Temperature data from HDS units and handling of unknown sensor types

### PR DESCRIPTION
**Pull request by christomaszewski**
* Implemented parsing of Temperature data from HDS units.
* Added handling of unknown sensor types.

### CODE REVIEW

File: app/src/main/java/com/platypus/android/server/VehicleServerImpl.java [hunk 1]
```diff
@@ -423,6 +423,7 @@ protected void onCommand(JSONObject cmd) {
                                 reading.data = new double[]{ecData, tempData};
                             } catch (NumberFormatException e) {
                                 Log.w(TAG, "Received malformed ES2 Sensor Data: " + value);
+                                continue;
```
Comment (psigen) [inline app/src/main/java/com/platypus/android/server/VehicleServerImpl.java:426]: Good catch, we don't want to transmit in these cases.

**Comment by psigen** [comment]
Looks pretty good to me, if it passes testing over the weekend I am fine with merging it.

File: app/src/main/java/com/platypus/android/server/VehicleServerImpl.java [hunk 2]
```diff
@@ -479,15 +489,23 @@ protected void onCommand(JSONObject cmd) {
...
-                        sendSensor(sensor, reading);
+                        // Attempt to log the collected sensor reading
                         mLogger.info(new JSONObject()
                                 .put("sensor", new JSONObject()
                                         .put("channel", reading.channel)
                                         .put("type", reading.type.toString())
                                         .put("data", new JSONArray(reading.data))));
+
+                        // Send out the collected sensor reading
+                        sendSensor(sensor, reading);
```
Comment (psigen) [inline app/src/main/java/com/platypus/android/server/VehicleServerImpl.java:508]: Wait, were we just not sending any sensor data before?
How was this working?

Comment (christomaszewski) [reply app/src/main/java/com/platypus/android/server/VehicleServerImpl.java:508]: just reordered the sending to be after logging so if there are further crashes we have the string logged.

**Comment by christomaszewski** [comment]
We've been using this successfully in Cololmbia with no issues. I'll push it out to UCM who initially had the error this was made to fix. I think we can merge.

**Pull request merged**